base/complex.jl

```diff
@@ -996,7 +996,8 @@ function round(z::Complex, rr::RoundingMode=RoundNearest, ri::RoundingMode=rr; k
     Complex(round(real(z), rr; kwargs...),
             round(imag(z), ri; kwargs...))
 end
-
+round(::Type{Complex{T}}, x::Complex) where {T<:Real} = Complex{T}(round(T, x.re), round(T, x.im))
+round(::Type{Complex{T}}, x::U) where U<:Union{Integer,AbstractFloat} where T<:Real = Complex{T}(round(x))
```

Comment:
Why not just `x::Real`?

Reply:
And shouldn't it call `round(T, x)`?

Reply:
not just `x::Real` because it seems to have ambiguities with `round(::Type{T}, x::Rational)`.

Reply:
You fix ambiguities by adding a more specific method, not by removing a more general method.

Reply:
Isn't `Complex{T}(round(x))` and `Complex{T}(round(T,x))` the same thing @stevengj ?

Reply:
I will fix this then :+1: